### ISSUE TITLE
[Bug] Behavior 'AllowInjectionScopeLocation' doesn't work with 'ConstructorSelectionMethod.Dynamic' behavior

### ISSUE THREAD

**Issue by mcdis** [ipjohnson/Grace#171]
test:
```
class Q
{
  public Q(IInjectionScope _scope) {}
}
```

```
var container = new DependencyInjectionContainer(_ =>
{
  _.Behaviors.ConstructorSelection = ConstructorSelectionMethod.Dynamic;
  _.Behaviors.AllowInjectionScopeLocation = true;
});
var q = container.Locate<Q>();
```

**Comment by ipjohnson** [comment]
That does sound like a bug. I’ll see if I can find some time to look at this in the next few days.

**Comment by ipjohnson** [comment]
I've checked in a change for this issue. I have one more thing that I want to address for version 6.4.0. I'll try and get a new version out to nuget this weekend. Till then you can try the nightly nuget feed below

https://ci.appveyor.com/nuget/grace-master

**Comment by ipjohnson** [comment]
I've released 6.4.0-RC685 with this fix
